py/dml/codegen.py

```diff
@@ -2016,7 +2039,7 @@ def stmt_while(stmt, location, scope):
     if stmt.site.dml_version() == (1, 2) and cond.constant and not cond.value:
         return [mkNull(stmt.site)]
     else:
-        loop_stack.append('c')
+        loop_stack.append(CLoopContext())
         try:
```

Comment:
Perhaps a context manager would make sense here, i.e. `with CLoopContext():`, like we do with `codegen.Failure`.

Reply:
That does sound reasonable.